File: Solutions/Day 5: Linked List/5.3_merge_ll.cpp
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) 
    {
        ListNode * res = NULL;
        if(list1 == NULL) {
            return list2;
        }
        else if(list2 == NULL) {
            return list1;
        }
        
        if(list1->val > list2->val) {
            res = list2;
            res->next = mergeTwoLists(list1, list2->next);
        }
        else {
            res = list1;
            res->next = mergeTwoLists(list1->next, list2);
        }
        return res;
    }   
};

File: Solutions/Day 5: Linked List/5.3_merge_ll.py
class Solution:
    def mergeTwoLists(self, list1, list2):
        res = None
        if list1 is None:
            return list2
        if list2 is None:
            return list1
        
        if list1.val > list2.val:
            res = list2
            res.next = self.mergeTwoLists(list1, list2.next)
        else:
            res = list1
            res.next = self.mergeTwoLists(list1.next, list2)
        return res
